client/lib/domains/constants.js

```diff
@@ -17,7 +17,40 @@ const registrar = {
 	MAINTENANCE: 'Registrar TLD Maintenance'
 };
 
+const tlds = {
```

Comment:
I didn't meant we use that code as is, we should use `QueryProductsList`,
I pasted the code in a manner it'll be easier to test in console =)

We need only to create a memoizing selector for the products map we'll have at the redux store:

```
	const tlds = {};

	Object.keys( productsMap )
	.map( key => productsMap[ key ] )
	.filter( product => product.is_domain_registration )
	.map( product => ( {
		tld: product.product_name.indexOf( '.' ) > -1 ? product.product_name.match( /\.([^ ]+)/ )[ 1 ] : 'DEFAULT',
		product_slug: product.product_slug
	} ) )
	.forEach( product => tlds[ product.tld ] = product.product_slug )
	
	tlds[ 'com' ] = tlds[ 'net' ] = tlds[ 'org' ] = tlds.DEFAULT;

	delete tlds.DEFAULT;

	return tlds;
```